front/src/modules/powerRestriction/hooks/usePowerRestrictionSelectorBehaviours.ts

```diff
@@ -153,6 +168,22 @@ const usePowerRestrictionSelectorBehaviours = ({
 
     if (fromPathStep && toPathStep) {
       dispatch(deletePowerRestrictionRanges({ from: fromPathStep, to: toPathStep }));
+      // clean customRanges if the deleted range was the only non-empty one
+      if (
+        ranges.length === 3 &&
+        ranges.reduce((acc, range) => (range.value !== NO_POWER_RESTRICTION ? acc + 1 : acc), 0) ===
+          1
```

Comment:
nit : we could also do
```suggestion
        ranges.filter((range) => range.value !== NO_POWER_RESTRICTION).length === 1
```

Reply:
Nice catch :pray: